modules/textInput/TextInput.tsx

```diff
@@ -158,26 +179,42 @@ export const TextInput: React.FC<PropsWithChildren<
           {label}
         </span>
       </div>
-      <input
-        {...props}
-        disabled={disabled}
-        ref={inputRef}
-        className={cx(styles.input, disabled && "disabled")}
-        onChange={e => {
-          if (onChange) {
-            onChange(e);
-          }
-          setHasContent(e.target.value.length > 0);
-        }}
-        onBlur={e => {
-          setIsFocused(false);
-          return props.onBlur?.(e);
-        }}
-        onFocus={e => {
-          setIsFocused(true);
-          return props.onFocus?.(e);
-        }}
-      />
+      <div className={cx(styles.inputContainer)}>
+        {prefixIcon && (
+          <Icon
+            name={prefixIcon as IconNames}
+            size={16}
+            className={cx(styles.prefixIcon)}
+          />
+        )}
+        <input
+          {...props}
+          disabled={disabled}
+          ref={inputRef}
+          className={cx(styles.input, disabled && "disabled")}
+          onChange={e => {
+            if (onChange) {
+              onChange(e);
+            }
+            setHasContent(e.target.value.length > 0);
+          }}
+          onBlur={e => {
+            setIsFocused(false);
+            return props.onBlur?.(e);
+          }}
+          onFocus={e => {
+            setIsFocused(true);
+            return props.onFocus?.(e);
+          }}
+        />
+        {suffixIcon && (
+          <Icon
```

Comment:
I think we could start using the `Icon` component from the registry since on ddm it'll probably warn / error us that only the icons on diana are available to pass through prop on suffixIcon / prefixIcon
what do you think ?

Reply:
you're correct. Need to check if it is registering the Icon